ccaas_builder/cmd/build/main.go

```diff
@@ -0,0 +1,175 @@
+	metadataFile := filepath.Clean(filepath.Join(metadataDir, "metadata.json"))
+	connectionDestFile := filepath.Join(outputDir, "/connection.json")
+	metainfoSrcDir := filepath.Join(sourceDir, "META-INF")
+	metainfoDestDir := filepath.Join(outputDir, "META-INF")
+
+	// Process and check the metadata file, then copy to the output location
+	metadataFileContents, err := ioutil.ReadFile(metadataFile)
+	if err != nil {
+		return err
+	}
+
+	var metadata ChaincodeMetadata
+	if err := json.Unmarshal(metadataFileContents, &metadata); err != nil {
+		return err
+	}
+
+	if strings.ToLower(metadata.Type) != "ccaas" {
+		return fmt.Errorf("chaincode type should be ccaas, it is %s", metadata.Type)
+	}
+
+	if err := copy.Copy(metadataDir, outputDir); err != nil {
+		return fmt.Errorf("failed to copy build metadata folder: %s", err)
+	}
+
+	if _, err := os.Stat(metainfoSrcDir); !os.IsNotExist(err) {
+		if err := copy.Copy(metainfoSrcDir, metainfoDestDir); err != nil {
+			return fmt.Errorf("failed to copy build META-INF folder: %s", err)
+		}
+	}
+
+	// Process and update the connections file
+	fileInfo, err := os.Stat(connectionSrcFile)
+	if err != nil {
+		return fmt.Errorf("connection.json not found in source folder: %s", err)
+	}
+
+	connectionFileContents, err := ioutil.ReadFile(connectionSrcFile)
+	if err != nil {
+		return err
+	}
+
+	// read the connection.json file into structure to process
+	var connectionData Connection
+	if err := json.Unmarshal(connectionFileContents, &connectionData); err != nil {
+		return err
+	}
+
+	// Treat each of the string fields in the connection.json as Go template
+	// strings. They can be fixed strings, but if they are templates
+	// then the JSON string that is defined in CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG
+	// is used as the 'context' to parse the string
+
+	updatedConnection := Connection{}
+	var cfg map[string]interface{}
+
+	cfgString := os.Getenv("CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG")
+	if cfgString != "" {
+		if err := json.Unmarshal([]byte(cfgString), &cfg); err != nil {
+			return fmt.Errorf("Failed to unmarshal %s", err)
+		}
+	}
+
+	updatedConnection.Address, err = execTempl(cfg, connectionData.Address)
+	if err != nil {
+		return fmt.Errorf("Failed to parse the Address field template: %s", err)
+	}
+
+	updatedConnection.DialTimeout, err = execTempl(cfg, connectionData.DialTimeout)
+	if err != nil {
+		return fmt.Errorf("Failed to parse the DialTimeout field template: %s", err)
+	}
+
+	if connectionData.TLS {
+		updatedConnection.TLS = true
+		updatedConnection.ClientAuth = connectionData.ClientAuth
+
+		updatedConnection.RootCert, err = execTempl(cfg, connectionData.RootCert)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the RootCert field template: %s", err)
+		}
+		updatedConnection.ClientKey, err = execTempl(cfg, connectionData.ClientKey)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the ClientKey field template: %s", err)
+		}
+
+	} else {
```

Comment:
No need for an else here.

Reply:
Can you expand on the thinking here please?  There are two codepaths, one if the `connectionData.TLS` is true and one if it's false.

Reply:
In the If statement you'll do a return with the error.
If you don't fall into the If statement, it will continue on the normal code path. No else statement is needed for the normal code path.

Reply:
Both branches are normal code paths here... 

```
if (tlsEnabled) {
// stuff for tls
} else {
// stuff for non tls
}
```

The only other way this could be written would be 
```
/// assume non-tls and setup tls stuff
if (tlsEnabed){
 // setup tls
}
```

Reply:
Sorry, my second response was incorrect, I was thinking of the other occurrences.

I stand by the initial review comment however. In Go the zero value of a boolean is false. Therefore you don't need to explicitly set `updatedConnection.TLS` to false. You can simply check for the true case in the If statement and completely avoid the else. This makes the code easier to read.

Many of my comments are simply Go conventions that may seem a bit arbitrary. But when a whole community of developers share the same practices the code becomes more readable by that community!